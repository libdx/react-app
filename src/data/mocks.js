//@flow

import type { Movie } from '../movie'

export const forrest: Movie = {
    imageURL: "http://t0.gstatic.com/images?q=tbn:ANd9GcSppDgk99BKVA4TJtWc1FN4-HUkdWrFNfMm1-M0nQ01sIOcbTZu",
    title: "Forrest Gump",
    releaseYear: "1994",
    genre: "Comedy-drama",
    description: "Slow-witted Forrest Gump (Tom Hanks) has never thought of himself as disadvantaged, and thanks to his supportive mother (Sally Field), he leads anything but a restricted life. Whether dominating on the gridiron as a college football star, fighting in Vietnam or captaining a shrimp boat, Forrest inspires people with his childlike optimism. But one person Forrest cares about most may be the most difficult to save -- his childhood love, the sweet but troubled Jenny (Robin Wright)."
}

export const movies = [forrest, forrest, forrest, forrest, forrest, forrest]

