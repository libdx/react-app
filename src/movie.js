type Movie = {
    imageURL: string,
    title: string,
    releaseYear: string,
    gener: string,
    description: string,
}
