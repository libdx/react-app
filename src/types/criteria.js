
export const criteria = {
    TITLE: "title",
    GENRE: "genres"
}

export type Criteria = $Keys<typeof criteria>

