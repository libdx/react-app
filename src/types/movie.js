//@flow

export type Movie = {
    id: number,
    title: string,
    tagline: string,
    vote_average: number,
    vote_count: number,
    release_date: string,
    poster_path: string,
    overview: string,
    budget: number,
    revenue: number,
    runtime: number,
    genres: Array<string>,
}

export const movieID = (movie: Movie): number => {
    return movie.id
}

export const mostAppropriateGenre = (movie: Movie): ?string => {
    return movie.genres[0]
}

export const genre = (movie: Movie): string => {
    return movie.genres.join(', ')
}

export const title = (movie: Movie): string => {
    return `${movie.title} (${movie.vote_average})`
}

export const releaseYear = (movie: Movie): number => {
    const date = new Date(movie.release_date)
    return date.getFullYear()
}

export const overview = (movie: Movie): string => {
    return movie.overview
}

export const posterURL = (movie: Movie): string => {
    return movie.poster_path
}

export const date = (movie: Movie): Date => {
    return new Date(movie.release_date)
}

export const raiting = (movie: Movie): number => {
    return movie.vote_average
}
