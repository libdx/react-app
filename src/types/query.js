export type Query = {
    term: string,
    searchScope: SearchScope,
}

