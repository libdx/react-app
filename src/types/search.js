
export const searchScopes = {
    TITLE: "title",
    GENRE: "genres"
}

export type SearchScope = $Keys<typeof searchScopes>

