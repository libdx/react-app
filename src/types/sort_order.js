export const sortOrder = {
    NONE: 'none',
    RELEASE_DATE: 'release-date',
    RATING: 'rating'
}

export type SortOrder = $Keys<typeof sortOrder>

