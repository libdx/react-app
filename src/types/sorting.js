export const sortKeys = {
    RELEASE_DATE: 'release_date',
    RATING: 'vote_average'
}

export type SortKey = $Keys<typeof sortKeys>

export const sortOrders =  {
    ASC: 'asc',
    DESC: 'desc',
}

export type SortOrder = $Key<typeof sortOrders>

