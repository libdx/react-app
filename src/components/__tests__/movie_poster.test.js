import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MoviePoster from '../movie_poster';
import { posterURL } from '../../types/movie';

// FIXME: fire out how to properly mock model objects
import { forrest as movie } from '../../data/mocks';

describe('MoviePoster', () => {
  let element;
  let component;

  beforeEach(() => {
    element = <MoviePoster movie={movie} />;
    component = shallow(element);
  });

  it('has correct URL', () => {
    const image = component.find('.poster-img');
    expect(image.props().src).toEqual(posterURL(movie));
  });

  it('renders correctly', () => {
    const tree = renderer.create(element).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
