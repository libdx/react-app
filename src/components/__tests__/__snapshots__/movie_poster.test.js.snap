// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePoster renders correctly 1`] = `
<img
  className="card-img-top poster-img"
  src="https://image.tmdb.org/t/p/w500/yE5d3BUhE8hCnkMUJOo1QDoOGNz.jpg"
/>
`;
